### PR TITLE
Refactor Application to use TruthTable instead of AddressBook

### LINKED ISSUE #172: Change Instances of AddressBook to TruthTable

### PR DESCRIPTION
**Pull request by Devanshshah1309**
Fixes #172

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,5 +33,5 @@ Then, **TruthTable** is _perfect_ for you.
 
 # Acknowledgements
 
-This project is based on the `AddressBook-Level3` project created by the [SE-EDU initiative](https://se-education.org).
+This project is based on the `TruthTable-Level3` project created by the [SE-EDU initiative](https://se-education.org).
```
Comment (mjgui) [inline README.md:36]: change back pls

File: src/main/java/seedu/address/storage/JsonSerializableTruthTable.java [hunk 2]
```diff
@@ -9,73 +9,73 @@
...
+import seedu.address.model.ReadOnlyTruthTable;
+import seedu.address.model.TruthTable;
 import seedu.address.model.person.Person;
 import seedu.address.model.team.Team;
 
 /**
- * An Immutable AddressBook that is serializable to JSON format.
+ * An Immutable TruthTable that is serializable to JSON format.
  */
 @JsonRootName(value = "addressbook")
```
Comment (changjy44) [inline src/main/java/seedu/address/storage/JsonSerializableTruthTable.java:20]: Should this be addressbook? Can check compatability first.
```suggestion
@JsonRootName(value = "truthtable")
```

File: src/test/java/seedu/address/testutil/TruthTableBuilder.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
+package seedu.address.testutil;
+
+import seedu.address.model.TruthTable;
+import seedu.address.model.person.Person;
+
+/**
+ * A utility class to help with building Addressbook objects.
```
Comment (changjy44) [inline src/test/java/seedu/address/testutil/TruthTableBuilder.java:7]:
```suggestion
 * A utility class to help with building Truthtable objects.
```

**Review by changjy44 — APPROVED** [review]
LGTM, some minor edits. Nice work for the refactoring.

File: src/main/java/seedu/address/model/TruthTable.java [hunk 4]
```diff
@@ -37,23 +37,23 @@ public class AddressBook implements ReadOnlyAddressBook {
...
+     * Creates an TruthTable using the Persons in the {@code toBeCopied}
      */
-    public AddressBook(ReadOnlyAddressBook toBeCopied) {
+    public TruthTable(ReadOnlyTruthTable toBeCopied) {
         this();
         resetData(toBeCopied);
     }
 
     /**
      * Initialises a new Addressbook with a default team.
```
Comment (changjy44) [inline src/main/java/seedu/address/model/TruthTable.java:52]:
```suggestion
     * Initialises a new Truthtable with a default team.
```

File: docs/team/johndoe.md [hunk 5]
```diff
@@ -3,9 +3,9 @@ layout: page
 title: John Doe's Project Portfolio Page
 ---
 
-### Project: AddressBook Level 3
+### Project: TruthTable Level 3
 
-AddressBook - Level 3 is a desktop address book application used for teaching Software Engineering principles. The user interacts with it using a CLI, and it has a GUI created with JavaFX. It is written in Java, and has about 10 kLoC.
+TruthTable - Level 3 is a desktop TruthTable application used for teaching Software Engineering principles. The user interacts with it using a CLI, and it has a GUI created with JavaFX. It is written in Java, and has about 10 kLoC.
```
Comment (mjgui) [inline docs/team/johndoe.md:8]: we can just delete this file tbh

File: docs/tutorials/TracingCode.md [hunk 6]
```diff
@@ -120,14 +120,14 @@ Recall from the User Guide that the `edit` command has the format: `edit INDEX [
```
Comment (mjgui) [inline docs/tutorials/TracingCode.md:120]: For that matter, can we just delete all the files in `docs/tutorials`?

File: src/main/java/seedu/address/commons/core/index/Index.java [hunk 7]
```diff
@@ -2,7 +2,7 @@
 
 /**
  * Represents a zero-based or one-based index.
- *
+ * <p>
```
Comment (mjgui) [inline src/main/java/seedu/address/commons/core/index/Index.java:5]: ?

File: src/main/java/seedu/address/logic/commands/EditLinkCommand.java [hunk 8]
```diff
@@ -41,7 +41,7 @@ public class EditLinkCommand extends Command {
                     + " \"Google\" " + FLAG_URL_STR + " https://google.com ";
     public static final String MESSAGE_EDIT_LINK_SUCCESS = "Edited link: %1$s";
     public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
-    public static final String MESSAGE_DUPLICATE_LINK = "This link already exists in the address book.";
+    public static final String MESSAGE_DUPLICATE_LINK = "This link already exists in the TruthTable.";
```
Comment (mjgui) [inline src/main/java/seedu/address/logic/commands/EditLinkCommand.java:44]:
```suggestion
    public static final String MESSAGE_DUPLICATE_LINK = "This link already exists in the team.";
```

File: src/main/java/seedu/address/model/TruthTable.java (hunk 4 above)
Comment (mjgui) [reply src/main/java/seedu/address/model/TruthTable.java:52]: You should do a non case-sensitive search for "addressbook" i think, from what i remember there's a lot of random stuff

File: src/main/java/seedu/address/model/person/UniquePersonList.java [hunk 9]
```diff
@@ -17,7 +17,7 @@
  * persons uses Person#isSamePerson(Person) for equality so as to ensure that the person being added or updated is
  * unique in terms of identity in the UniquePersonList. However, the removal of a person uses Person#equals(Object) so
  * as to ensure that the person with exactly the same fields will be removed.
- *
+ * <p>
```
Comment (mjgui) [inline src/main/java/seedu/address/model/person/UniquePersonList.java:20]: ?

File: src/main/java/seedu/address/model/team/UniqueLinkList.java [hunk 10]
```diff
@@ -14,7 +14,7 @@
 /**
  * A list of links that enforces uniqueness between its elements and does not allow nulls.
  * A link is considered unique solely based on the name of the team.
- *
+ * <p>
```
Comment (mjgui) [inline src/main/java/seedu/address/model/team/UniqueLinkList.java:17]: ?

File: src/main/java/seedu/address/ui/HelpWindow.java [hunk 11]
```diff
@@ -46,21 +46,21 @@ public HelpWindow() {
...
+     *                               <li>
+     *                               if this method is called during animation or layout processing.
+     *                               </li>
+     *                               <li>
+     *                               if this method is called on the primary stage.
+     *                               </li>
+     *                               <li>
+     *                               if {@code dialogStage} is already showing.
+     *                               </li>
+     *                               </ul>
```
Comment (mjgui) [inline src/main/java/seedu/address/ui/HelpWindow.java:63]: can we undo this? it's a lot less readable now

File: README.md (hunk 1 above)
Comment (Devanshshah1309) [inline README.md:36]:
```suggestion
This project is based on the `AddressBook-Level3` project created by the [SE-EDU initiative](https://se-education.org).
```

File: docs/index.md [hunk 12]
```diff
@@ -8,10 +8,10 @@ title: TruthTable
...
-**AddressBook is a desktop application for managing your contact details.** While it has a GUI, most of the user interactions happen using a CLI (Command Line Interface).
+**TruthTable is a desktop application for managing your contact details.** While it has a GUI, most of the user interactions happen using a CLI (Command Line Interface).
 
-* If you are interested in using AddressBook, head over to the [_Quick Start_ section of the **User Guide**](UserGuide.html#quick-start).
-* If you are interested about developing AddressBook, the [**Developer Guide**](DeveloperGuide.html) is a good place to start.
+* If you are interested in using TruthTable, head over to the [_Quick Start_ section of the **User Guide**](UserGuide.html#quick-start).
+* If you are interested about developing TruthTable, the [**Developer Guide**](DeveloperGuide.html) is a good place to start.
 
 
 **Acknowledgements**
```
[the remark this answers is not in the document]
Comment (Devanshshah1309) [reply docs/index.md:17]: Probably yeah can do in another PR

File: docs/tutorials/TracingCode.md (hunk 6 above)
Comment (Devanshshah1309) [reply docs/tutorials/TracingCode.md:120]: Why though? We can take inspiration from it for our UG later right?

**Pull request merged**